### ISSUE TITLE
Multiple .NET 5 targets breaks ResolvePackageDependencies task with duplicate dictionary key exception

### ISSUE THREAD

**Issue by drewnoakes** [dotnet/sdk#15291]
SDK 5.0.200-preview.20614.14

Given this project, which targets both `net5.0` and `net5.0-windows`, the design-time build fails.

```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFrameworks>net5.0;net5.0-windows</TargetFrameworks>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="MetadataExtractor" Version="2.4.0" />
  </ItemGroup>
</Project>
```

The exception is:

```
C:\Program Files\dotnet\sdk\5.0.200-preview.20614.14\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(186,5):

error MSB4018: The "ResolvePackageDependencies" task failed unexpectedly.

System.ArgumentException: An item with the same key has already been added.
   at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   at Microsoft.NET.Build.Tasks.ResolvePackageDependencies.ExecuteCore()
   at Microsoft.NET.Build.Tasks.TaskBase.Execute()
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()
```

Looking at the relevant code, it seems that both `net5.0` and `net5.0-windows` have the same value for `DotNetFrameworkName`. It seems a different identifier should be used, which encompasses the platform details as well.

There are two `ToDictionary` calls in this method, so it may be the other that's causing the issue.

https://github.com/dotnet/sdk/blob/bfa0472d650030a087cbea9361df16791346094d/src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageDependencies.cs#L153-L166

**Comment by AraHaan** [comment]
net5.0-windows is an extension to net5.0 which I think it considers it to be almost the same thing (just that it adds the ability to specify that you want it to use Windows only APIs with it as well).

**Comment by sfoslund** [comment]
Fixed in 5.0.2xx by https://github.com/dotnet/sdk/pull/15369
